kv/kvserver: TestStoreRangeMergeWithData failed

kv/kvserver.TestStoreRangeMergeWithData [failed](https://tanzanite.cluster.engflow.com/invocations/default/1c20c06e-9857-4f3b-affe-878eed3b1177?testReportRun=2&testReportShard=35&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [361f58bdb476e76fd3c6c9a2f54a48b7c74e9797](https://github.com/cockroachdb/cockroach/commits/361f58bdb476e76fd3c6c9a2f54a48b7c74e9797):

Fatal error:

```
panic: test timed out after 1h16m37s
running tests:
	TestStoreRangeMergeWithData (1h16m37s)
	TestStoreRangeMergeWithData/retries=0 (1h16m37s)

```
Stack: 

```
goroutine 921659 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestStoreRangeMergeWithData
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestStoreRangeMergeWithData2661097904
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestStoreRangeMergeWithData/retries=0
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=35</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #125779 kv/kvserver: TestStoreRangeMergeWithData failed [A-testing C-bug C-test-failure O-robot P-3 T-kv branch-release-24.1.2-rc]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeWithData.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39820

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -264,8 +264,6 @@ func TestStoreRangeMergeWithData(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 126249)
-
 	for _, retries := range []int64{0, 3} {
 		t.Run(fmt.Sprintf("retries=%d", retries), func(t *testing.T) {
 			mergeWithData(t, retries)
@@ -275,6 +273,10 @@ func TestStoreRangeMergeWithData(t *testing.T) {
 
 func mergeWithData(t *testing.T, retries int64) {
 	ctx := context.Background()
+
+	// Set a long txn liveness threshold so that the merge txn cannot be aborted,
+	// even when we manually advance the clock to trigger a lease acquisition.
+	defer txnwait.TestingOverrideTxnLivenessThreshold(time.Hour)()
 
 	manualClock := hlc.NewHybridManualClock()
 	var store *kvserver.Store

